VERSIONING.md

```diff
@@ -0,0 +1,119 @@
+# Versioning
+
+This document aims to provide an overview of how VAST is versioned, and what
+significance a version bump carries.
+
+Run the `vast version` command to get VAST's current version, and the versions
+of its major dependencies that it currently uses:
+
+```json
+{
+  "VAST": "v1.0.4-61-geed8cb41a3",
+  "VAST Build Tree Hash": "0eb048d0eeda5be7081f0c3a659ff17f",
+  "CAF": "0.17.6",
+  "Apache Arrow": "6.0.1",
+  "jemalloc": "5.2.1",
+  "plugins": {
+    "compaction": "v1.2.0-61-g9131346958",
+    "matcher": "v2.1.4-23-gfb69e84ade",
+  }
+}
+```
+
+## Version Components
+
+VAST's version number consists of three mandatory and two optional parts:
```

Comment:
The reader immediately wonders what the versioning scheme relates to most. We should mention upfront that it's SemVer'ish.